packages/compass-explain-plan/src/modules/explain.js

```diff
@@ -291,6 +318,7 @@ export function isAggregationExplainOutput(explainOutput) {
 
 /**
  * Fetches the explain plan.
+ * TODO: Declutter this middleware as it is getting difficult to read
```

Comment:
Makes sense - also I didn't do justice in explaining what that todo was meant to be.

I thought that it would be much clearer if this middleware would dispatch explicit actions like `explainPlanFetched`, `explainPlanFetchFailed`, `explainPlanParseFailed` (just some examples) instead of always dispatching `explainPlanFetched` with nested properties of explain modified in different cases. It was not clear to understand upfront what state certain properties (being changed) were related to.